backend/src/test/java/com/redhat/cloud/notifications/routers/NotificationResourceTest.java

```diff
@@ -330,6 +331,111 @@ void testEventTypeFetchingByBundleApplicationAndEventTypeName() {
+        List<String> unmutedEventTypeNames = List.of(String.format(TEST_EVENT_TYPE_FORMAT, 0), String.format(TEST_EVENT_TYPE_FORMAT, 1), String.format(TEST_EVENT_TYPE_FORMAT, 14));
+        behaviorGroupRepository.updateEventTypeBehaviors(ORG_ID, eventTypes.stream().filter(ev -> ev.getName().equals(unmutedEventTypeNames.getFirst())).findFirst().get().getId(), Set.of(behaviorGroupId1));
+        behaviorGroupRepository.updateEventTypeBehaviors(ORG_ID, eventTypes.stream().filter(ev -> ev.getName().equals(unmutedEventTypeNames.get(1))).findFirst().get().getId(), Set.of(behaviorGroupId1));
+        behaviorGroupRepository.updateEventTypeBehaviors(ORG_ID, eventTypes.stream().filter(ev -> ev.getName().equals(unmutedEventTypeNames.get(2))).findFirst().get().getId(), Set.of(behaviorGroupId2));
+
+        String response = given()
+                .when()
+                .header(identityHeader)
+                .queryParam("excludeMutedTypes", "true")
+                .get("/notifications/eventTypes")
+                .then()
+                .statusCode(200)
+                .contentType(JSON)
+                .extract().asString();
+
+        JsonObject page = new JsonObject(response);
+        JsonArray respEventTypes = page.getJsonArray("data");
+        ArrayList<String> respEventTypeNames = new ArrayList<>();
+        for (int i = 0; i < respEventTypes.size(); i++) {
+            JsonObject ev = respEventTypes.getJsonObject(i);
+            respEventTypeNames.add(ev.getString("name"));
+        }
+
+        assertTrue(unmutedEventTypeNames.containsAll(respEventTypeNames) && respEventTypeNames.containsAll(unmutedEventTypeNames));
+        assertFalse(respEventTypeNames.contains(String.format(TEST_EVENT_TYPE_FORMAT, 2)));
+        assertEquals(3, page.getJsonObject("meta").getInteger("count"));
+    }
+
+    @Test
+    void testEventTypeFetchingByBundleApplicationEventTypeNameAndExcludeMutedTypes() {
+        UUID bundleId = helpers.createTestAppAndEventTypes();
+        List<Application> apps = applicationRepository.getApplications(TEST_BUNDLE_NAME);
+        Application app1 = apps.stream().filter(a -> a.getName().equals(TEST_APP_NAME)).findFirst().get();
+        UUID appId1 = app1.getId();
+        Application app2 = apps.stream().filter(a -> a.getName().equals(TEST_APP_NAME_2)).findFirst().get();
+        UUID appId2 = app2.getId();
+        Header identityHeader = initRbacMock(ACCOUNT_ID, ORG_ID, "user", FULL_ACCESS);
+        UUID behaviorGroupId1 = helpers.createBehaviorGroup(ACCOUNT_ID, ORG_ID, "behaviour-group-1", bundleId).getId();
+        UUID behaviorGroupId2 = helpers.createBehaviorGroup(ACCOUNT_ID, ORG_ID, "behaviour-group-2", bundleId).getId();
+
+        // bgroup1 assigned to ev0 and ev1 on TEST_APP_NAME, bgroup2 assigned to ev1 on TEST_APP_NAME_2, all other event types unassigned
+        UUID endpointId1 = helpers.createWebhookEndpoint(ACCOUNT_ID, ORG_ID);
```

Comment:
You can skip creating webhook endpoints and link them to the behavior groups in this test method as well.